aspnetcore/mvc/models/file-uploads/samples/3.x/SampleApp/Utilities/FileHelpers.cs

```diff
@@ -208,7 +201,7 @@ private static bool IsValidFileExtensionAndSignature(string fileName, Stream dat
 
             data.Position = 0;
 
-            using (var reader = new BinaryReader(data))
+            using (var reader = new BinaryReader(data, System.Text.Encoding.UTF8, leaveOpen: true))
```

Comment:
If we're going to operate on a real stream, we should probably not close it after validation.